common/buildcraft/factory/tile/TileChute.java

```diff
@@ -0,0 +1,212 @@
+            slotIndexes = ((ISidedInventory) inventory).getSlotsForFace(side);
+        }
+        for(int j : slotIndexes) {
+            ItemStack stack = inventory.getStackInSlot(j);
+            if(StackUtil.canMerge(itemStack, stack) || stack == null) {
+                if(stack == null) {
+                    stack = itemStack.copy();
+                    stack.stackSize = 0;
+                }
+                int oldSize = stack.stackSize;
+                int maxTransfer = 1;
+                stack.stackSize = MathUtils.clamp(stack.stackSize + Math.min(itemStack.stackSize, maxTransfer), 0, itemStack.getMaxStackSize());
+                int used = stack.stackSize - oldSize;
+                itemStack.stackSize -= used;
+                if(stack.stackSize != 0) {
+                    inventory.setInventorySlotContents(j, stack);
+                } else {
+                    inventory.setInventorySlotContents(j, null);
+                }
+                break;
+            }
+        }
+        return itemStack;
+    }
+
+    private void putInInventory(EnumFacing side, IInventory inventory) {
+        for(int i = 0; i < 4; i++) {
+            if(inv.getStackInSlot(i) != null) {
+                ItemStack itemStack = inv.getStackInSlot(i).copy();
+                itemStack = putStackInInventory(side, inventory, itemStack);
+                inv.setStackInSlot(i, itemStack);
+            }
+        }
+    }
+
+    private void putInInventories(EnumFacing currentSide) {
+        List<EnumFacing> sides = Arrays.asList(EnumFacing.values());
+        Collections.shuffle(sides, new Random());
+
+        for(EnumFacing side : sides) {
+            if(side == currentSide) {
+                return;
+            }
+
+            IInventory inventory = getInventoryAtPosition(worldObj, pos.offset(side));
+
+            if(inventory == null) {
+                return;
+            }
+
+            if(isInventoryFull(inventory, side)) {
+                return;
+            }
+
+            putInInventory(side, inventory);
+        }
+    }
+
+    @Override
+    public void readFromNBT(NBTTagCompound nbt) {
+        super.readFromNBT(nbt);
+        progress = nbt.getInteger("progress");
+        battery.deserializeNBT(nbt.getCompoundTag("mj_battery"));
+    }
+
+    @Override
+    public NBTTagCompound writeToNBT(NBTTagCompound nbt) {
+        super.writeToNBT(nbt);
+        nbt.setInteger("progress", progress);
+        nbt.setTag("mj_battery", battery.serializeNBT());
+        return nbt;
+    }
+
+    // ITickable
+
+    @Override
+    public void update() {
+        if(worldObj.isRemote) {
+            return;
+        }
+
+        battery.tick(getWorld(), getPos());
+
+        // test with the output of a stone engine
+        battery.addPower(1000); // remove this
+
+        EnumFacing currentSide = (EnumFacing) worldObj.getBlockState(pos).getProperties().get(BlockBCBase_Neptune.BLOCK_FACING_6);
```

Comment:
You can use ````worldObj.getBlockState(pos).getValue(BlockBCBase_Neptune.BLOCK_FACING_6)```` instead. Although I have come across times when the tile still exists but the block doesn't, so you should have a check to make sure the block is the same.